files/scripts/syncd.sh

```diff
@@ -104,34 +104,32 @@ function stopplatform1() {
         debug "Stopped pmon service"
     fi
 
-    if [[ x$sonic_asic_platform != x"mellanox" ]] || [[ x$TYPE != x"cold" ]]; then
```

Comment:
>[[ x$TYPE != x"cold" ]]; [](http://example.com/codeflow?start=53&length=24)

Why remove the condition? Maybe you intend only to remove the sonic_asic_platform check.

Reply:
Hi @qiluo-msft , we intend to remove the whole if block because:

- if platform is not mellanox, it will trigger graceful shutdown flow no matter what type it is. It means that removing the condition won't affect any other vendor. 
- if platform is mellanox, it will trigger graceful shutdown if type is not cold (old logic). This is exactly what we try to fix. We move the `type != cold` check to Syncd.cpp, see PR https://github.com/sonic-net/sonic-sairedis/pull/1216 .

Reply:
@qiluo-msft  would you please check the feedback from Junchao?